modules/storage/cosmos-db/main.bicep

```
@@ -1,99 +1,353 @@
+param tables array = []
+
+@description('Array of role assignment objects that contain the "roleDefinitionIdOrName" and "principalId" to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, provide either the display name of the role definition, or its fully qualified ID in the following format: "/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11"')
+param roleAssignments array = []
+
+@description('The type of identity used for the Cosmos DB account. The type "SystemAssigned, UserAssigned" includes both an implicitly created identity and a set of user-assigned identities. The type "None" will remove any identities from the Cosmos DB account.')
+@allowed([ 'None', 'SystemAssigned', 'SystemAssigned,UserAssigned', 'UserAssigned' ])
+param identityType string = 'None'
+
+@description('The list of user-assigned managed identities. The user identity dictionary key references will be ARM resource ids in the form: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}"')
+param userAssignedIdentities object = {}
+
+@description('Private Endpoints that should be created for Azure Cosmos DB account.')
+param privateEndpoints array = []
+
+@description('List of key-value pairs that describe the resource.')
+param tags object = {}
+
+@allowed([
+  'CanNotDelete'
+  'NotSpecified'
+  'ReadOnly'
+])
+@description('Specify the type of lock on Cosmos DB account resource.')
+param lock string = 'NotSpecified'
+
+var varConsistencyPolicy = (defaultConsistencyLevel == 'BoundedStaleness') ? {
+  defaultConsistencyLevel: 'BoundedStaleness'
+  maxStalenessPrefix: maxStalenessPrefix
+  maxIntervalInSeconds: maxIntervalInSeconds
+} : {
+  defaultConsistencyLevel: defaultConsistencyLevel
 }
 
-var secondaryRegions = [for (region, i) in secondaryLocations: {
+var varSecondaryRegions = [for (region, i) in secondaryLocations: {
   locationName: contains(region, 'locationName') ? region.locationName : region
   failoverPriority: contains(region, 'failoverPriority') ? region.failoverPriority : i + 1
   isZoneRedundant: contains(region, 'isZoneRedundant') ? region.isZoneRedundant : isZoneRedundant
 }]
 
-var locations = union([
+var varLocations = union([
     {
       locationName: location
       failoverPriority: 0
       isZoneRedundant: isZoneRedundant
     }
-  ], secondaryRegions)
+  ],
+  enableServerless ? [] : varSecondaryRegions
+)
 
-var capabilities = union(
-  enableCassandra ? [ { name: 'EnableCassandra' } ] : [],
-  enableServerless ? [ { name: 'EnableServerless' } ] : []
+var varCapabilities = union(
+  capabilities,
+  enableServerless ? [ 'EnableServerless' ] : [],
+  backendApi == 'cassandra' ? [ 'EnableCassandra' ] : [],
+  backendApi == 'gremlin' ? [ 'EnableGremlin' ] : [],
+  backendApi == 'mongodb' ? [ 'EnableMongo' ] : [],
+  backendApi == 'table' ? [ 'EnableTable' ] : []
 )
```

```suggestion
var capabilityMappings = {
  cassandra: 'EnableCassandra'
  gremlin: 'EnableGremlin'
  mongodb: 'EnableMongo'
  table: 'EnableTable'
}
  
var varCapabilities = union(
  capabilities,
  enableServerless ? [ 'EnableServerless' ] : [],
  contains(capabilityMappings, backendApi) ? [ capabilityMappings[backendApi] ] : []
)
```

how about something like this to clean it up a little bit? Might be able to simplify that contains line even more.

---

agree, it looks a lot cleaner. Updated.